foyer/atomtyper.py

```diff
@@ -7,10 +7,37 @@
 from parmed import Structure
 
 from foyer.exceptions import FoyerError
+from foyer.smarts import SMARTS
 from foyer.smarts_graph import SMARTSGraph
 from foyer.topology_graph import TopologyGraph
 
 
+class AtomTypingRulesProvider:
+    """A generic rules provider for atomtyping, agnostic of the forcefield."""
```

Comment:
Can we get a docstring for these init parameters? Other than that, LGTM!

Reply:
Added